pm64/src/bgm/bgm_ron.rs

```diff
@@ -0,0 +1,655 @@
+        for (ron_track, bgm_track) in tracks_iter.zip(tracks.iter_mut()) {
+            *bgm_track = MaybeUninit::new(bgm::Track::from(ron_track));
+        }
+
+        Self {
+            pos: value.pos,
+            tracks: unsafe {
+                // SAFETY: the for loop above has initialised the array. This is also how the std
+                // documentation suggests initialising an array element-by-element:
+                // https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element
+                std::mem::transmute::<[MaybeUninit<bgm::Track>; 16], [bgm::Track; 16]>(tracks)
+            },
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Serialize, Deserialize, TypeDef)]
+#[serde(default)]
+pub struct Track {
+    #[serde(default)]
+    pub name: String,
+    pub is_disabled: bool,
+    pub polyphony: Polyphony,
+    pub is_drum_track: bool,
+    pub parent_track_idx: Option<u8>,
+    pub commands: RonCommandSeq,
+}
+
+impl Default for Track {
+    fn default() -> Self {
+        Self {
+            name: "".to_owned(),
+            is_disabled: true,
+            polyphony: Polyphony::Automatic,
+            is_drum_track: false,
+            parent_track_idx: None,
+            commands: Default::default(),
+        }
+    }
+}
+
+impl From<bgm::Track> for Track {
+    fn from(value: bgm::Track) -> Self {
+        let parent_track_idx = (value.parent_track_idx != 0).then(|| value.parent_track_idx - 1);
+
+        Self {
+            name: value.name,
+            is_disabled: value.is_disabled,
+            polyphony: Polyphony::from(value.polyphonic_idx),
+            is_drum_track: value.is_drum_track,
+            parent_track_idx,
+            commands: RonCommandSeq::from(&value.commands),
+        }
+    }
+}
+
+impl From<Track> for bgm::Track {
+    fn from(value: Track) -> Self {
+        let parent_track_idx = match value.parent_track_idx {
+            Some(value) => value + 1,
+            None => 0,
+        };
+
+        Self {
+            name: value.name,
+            is_disabled: value.is_disabled,
+            polyphonic_idx: u8::from(value.polyphony),
+            is_drum_track: value.is_drum_track,
+            parent_track_idx,
+            commands: CommandSeq::from(&value.commands),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize, TypeDef)]
+pub enum Polyphony {
+    Automatic,
+    Manual { voices: u8 },
+    Other { priority: u8 },
+}
+
+impl From<u8> for Polyphony {
+    fn from(value: u8) -> Self {
+        match value {
+            0 => Self::Manual { voices: 0 },
+            1 => Self::Manual { voices: 1 },
+            5 => Self::Manual { voices: 2 },
+            6 => Self::Manual { voices: 3 },
+            7 => Self::Manual { voices: 4 },
+            POLYPHONIC_IDX_AUTO_MAMAR => Self::Automatic,
+            _ => Self::Other { priority: value },
```

Comment:
Slightly unfortunate that the variants overlap eachother in u8 representation, but I guess that can't be helped